app/cdap/components/shared/ConfigurationGroup/index.tsx

```diff
@@ -37,6 +38,8 @@ import {
 import { IErrorObj } from 'components/shared/ConfigurationGroup/utilities';
 import { h2Styles } from 'components/shared/Markdown/MarkdownHeading';
 
+const COLLAPSIBLE_CONFIG_GROUPS = ['Advanced'];
```

Comment:
This gives us the ability to add any other config group to this list and make them collapsible. We could also add a `collapsible` prop to the plugins' json files, and use that to populate `initialConfigGroupExpandedState` on line 118, and also use it when rendering further down instead of line 274. I kinda like to keep the control here on the UI side but the only downside I can see is us relying on the `label` from the plugin's json file.

Reply:
yeah, this makes it inflexible. Providing control in the widget JSON with something like ``hideByDefault: true|false`` would be better.

Reply:
That's true but if we want to get this out for 6.7, I don't think we can apply the changes to the widget JSONs unless we push this back

Reply:
How big is that change to add the hidebyDefault and let widget jsons define it?

Reply:
On the UI it's not much of a change. I just make a small change in my PR.